parsers/ES.py

```diff
@@ -170,69 +171,55 @@ def fetch_production(
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> List[dict]:
+    check_valid_parameters(zone_key, session, target_datetime, logger)
     ses = session or Session()
     island_data = fetch_island_data(zone_key, ses, target_datetime)
-    data: List[Dict[str, Any]] = []
+    productionEventList = ProductionBreakdownList(logger)
 
-    # If we add more zones this should be turned into a map similar to ZONE_FlOORS mapping.
-    if zone_key == "ES-IB":
-        expected_range = {"coal": (50, 600)}
-    else:
-        expected_range = None
-
-    if island_data:
-        for response in island_data:
-            if response.production() >= 0:
-                response_data = {
-                    "datetime": get(response.timestamp).datetime,
-                    "production": {
-                        "biomass": response.waste,
-                        "coal": response.carbon,
-                        "gas": round((response.gas + response.combined), 2),
-                        "geothermal": 0.0,
-                        "hydro": response.hydraulic,
-                        "nuclear": 0.0,
-                        "oil": round((response.vapor + response.diesel), 2),
-                        "solar": response.solar,
-                        "unknown": response.other,
-                        "wind": response.wind,
-                    },
-                    "source": "demanda.ree.es",
-                    "storage": {"hydro": 0.0, "battery": 0.0},
-                    "zoneKey": zone_key,
-                }
-
-                ### Zone overrides
+    for event in island_data:
+        production = ProductionMix()
+        storage = StorageMix()
+        for mode in [
+            attribute
+            for attribute in getmembers(event)
+            if not attribute[0].startswith("__")
```

Comment:
I'm not a big fan of this. I adds some hidden logic and is not super future proof if we decide to change attributes in the future. Also I think part is not super readable.
Could you not just construct the mode directly through a mapping?

Reply:
Agreed, but due to the way the Response object in the Ree library is structured there it returns nothing to loop over.

I plan on creating a upstream PR to change this (and make it return None instead of 0 when there is no data).